### DISCUSSION TITLE
Is GPU support working?

### DISCUSSION THREAD

**Discussion by sorenwacker** [pycaret/pycaret#1318] [Q&A]
I wonder if the GPU support is really working with pycaret. I am trying to run it on a Kaggle node with GPU enabled. I installed pycaret 2.3. Usually, when I change XGBoost method to 'gpu_hist' I see a big change in the time required to fit a model. But I do not see that with pycaret. I setup the experiment like this:

```
exp_clf = setup(df, target = 'target', 
                train_size=0.9,
                pca=False,
                use_gpu=True,
                silent=True,
                fold=5,
                data_split_stratify=True,
                data_split_shuffle=True,
                fold_shuffle=True,
                fix_imbalance=True,
                polynomial_features=True,
                trigonometry_features=True,
                feature_selection=True,
                remove_multicollinearity=True,
                multicollinearity_threshold=0.99,
                transformation=True,
                normalize=True,
                normalize_method='robust'
               )

```

And then use `xgb = create_model('xgboost', tree_method = 'gpu_hist')` to fit the model.

However, I only see CPU usage and I also don't find the fitting time decreasing (subjectively).

**Comment by Yard1** [comment]
Can you try without `tree_method` in `create_model`? It should be applied automatically. I'll run some tests myself, perhaps something else is overwriting it.

**Reply by sorenwacker** [reply]
Yes, I tried that first. Did not seem to change anything.

**Reply by Yard1** [reply]
Thanks for the report, will investigate. Can you open an issue tracking this?

**Reply by sorenwacker** [reply]
Sure, just wanted to check in if other people have the same issue, or maybe I am doing something wrong.

**Reply by Yard1** [reply]
Hey @soerendip , I ran a test on Google Colab using the large Poker dataset, and there was a huge improvement in runtime when using xgboost with `use_gpu=True`.

https://colab.research.google.com/drive/10jD3VHDWrNAf_I7YoA57kIKLZ8yPQZZh?usp=sharing

Taking the results into account I think this may be an issue with Kaggle?

**Reply by sorenwacker** [reply]
Ok, maybe the dataset that I was using was too big. The create model command took a very long time in both modes and I did not see any activity on the GPU bar in the notebook. Thank you for checking.
